tmt/checks/__init__.py

```diff
@@ -223,6 +223,26 @@ def delegate(
             raw_data, logger
         )
 
+    @classmethod
+    def internal_checks(
+        cls,
+        logger: tmt.log.Logger,
+    ) -> list['Check']:
+        """
+        Create internal check data instances
+        """
+
+        from tmt.checks.internal import InternalCheck
+
+        checks: list[Check] = []
+
+        for check_plugin in _CHECK_PLUGIN_REGISTRY.iter_plugins():
+            plugin = cast(CheckPlugin[CheckT], check_plugin)
+            if issubclass(plugin._check_class, InternalCheck):
+                checks.append(cast(InternalCheck, plugin._check_class).create_default(logger))
```

Comment:
Probably rely on the duck typing, with `hasmethod(plugin._check_class, 'create_default')`. Or, add the same method to `Check` and let it return nothing, `InternalCheck` would then provide its own implementation.